Feature/connect to nuget feed

warrantor opened the pull request.

Commit:
Added sections

Commit:
Find the feed

xavierdecoster commented on package-managers/nuget.md:
```diff
@@ -1,25 +1,76 @@
+#### Public feed
+There are two types of public feeds you can connect to. A public feed you own and one from the gallery. The only real difference here, is where to find the URL of the feed.
+
+With this URL you can connect to the feed using your favorite tool, which could be [Visual Studio](#visual-studio), [JetBrains Rider](#jetbrains-rider) or the [NuGet-CLI](#nugetexe)
```
If we mention the NuGet CLI, we should use `NuGet CLI` (no dash) or `nuget.exe` (lowercased).
Also, there's more tools out there: `paket` and `dotnet CLI` are also relevant on this NuGet-specific page.

Perhaps rephrase to something like:

> MyGet feeds have multiple endpoints, and each one is specific to a particular package type and protocol. Pick the NuGet endpoint URL that matches the protocol version you want to use (e.g. `NuGet v3`), and use that to configure the feed endpoint in your favorite NuGet client.

xavierdecoster commented on package-managers/nuget.md:
```diff
@@ -1,25 +1,76 @@
+### Find your feed URL
+
+#### Public feed
+There are two types of public feeds you can connect to. A public feed you own and one from the gallery. The only real difference here, is where to find the URL of the feed.
```
Maybe combine the first two sentences?

> There are two types of public feeds you can connect to: public feed you own and one from the gallery. The only ...

maartenba replied:
I had never thought of it this way but you are right! There are two ways. Main focus should be on feeds you or your team own, though. Gallery is secondary (and good to have covered).

Plan is to have this "connect" dialog in the authenticated version one day, too.

xavierdecoster commented on package-managers/nuget.md:
```diff
@@ -1,25 +1,76 @@
+
+
+##### Gallery feed
+When connecting to a gallery feed you first click on the feed you want to connect to and there click on the "Connect to feed" button.
```
Preferably use `we` in docs, and highlight actions or button text in bold.

> When browsing the list of feeds on the [MyGet Gallery](https://www.myget.org/gallery) (or on the MyGet Enterprise Gallery), we can click on a particular feed to navigate to the feed's details page. When we click on the **Connect to feed** button, a dialog opens providing us with information on how to connect to this feed's endpoints.

maartenba replied:
Suggestion: `After browsing to a gallery feed, we can get its connection details using the **Connect to feed** button. This will open a dialog where we can find the various endpoints for this feed.`

xavierdecoster commented on package-managers/nuget.md:
```diff
@@ -1,25 +1,76 @@
+
+![Gallery feed details](assets/gallery-feed.png)
+
+A dialog opens where you will once again find your 'NuGet V3' URL.
```
> Copy the desired endpoint URL from this dialog.

maartenba replied:
`We will need the NuGet V3 URL, so let's copy it for later use.`

(think: you are sitting next to the person reading the docs, and both of you are walking through this and making the user succesful)

xavierdecoster commented on package-managers/nuget.md:
```diff
@@ -1,25 +1,76 @@
+
+![Gallery feed URL](assets/connect-gallery-feed.png)
+
+With this URL you can connect to the feed using your favorite tool, which could be [Visual Studio](#visual-studio), [JetBrains Rider](#jetbrains-rider) or the [NuGet-CLI](#nugetexe)
```
This paragraph is a duplicate of what you wrote [here](https://github.com/maartenba/myget-documentation/pull/1/files#diff-a6e799b6b0d7c47c3631837ea0698fdeR14).

However, I'd add a comment here with anchors to quickly navigate to the client-specific **Public Feed** sections below.

warrantor replied:
Should I repeat the hint in each point (e.g. public, gallery, private) or just ones in one of the points?

xavierdecoster commented on package-managers/nuget.md:
```diff
@@ -1,25 +1,76 @@
+With this URL you can connect to the feed using your favorite tool, which could be [Visual Studio](#visual-studio), [JetBrains Rider](#jetbrains-rider) or the [NuGet-CLI](#nugetexe)
+
+##### Private feed
+To consume private feeds you need to get consume permissions from the feed owner or admin. When you have these rights you can navigate to the 'feed details' where you can find the 'NuGet V3' URL. 
```
Use `we` form.

> Private feeds are only accessible to the feed owner, and to whomever the feed owner granted sufficient privileges. When we need to connect to a private feed we don't own, we may need to ask the feed owner to invite us to the feed.

TODO: add a link in the above section to the relevant docs article discussing "inviting users to a feed"

> If we own the feed, or already have sufficient privileges, we may navigate to the feed's details page on the MyGet web site, and select **Feed Details** in the vertical navigation menu on the left. The details view will show us the available feed endpoints.

xavierdecoster commented on package-managers/nuget.md:
```diff
@@ -1,25 +1,76 @@
+
+![Feed details](assets/feed-details.png)
+
+With this URL you can connect to the feed using your favorite tool, which could be [Visual Studio](#visual-studio), [JetBrains Rider](#jetbrains-rider) or the [NuGet-CLI](#nugetexe)
```
This paragraph is a duplicate of what you wrote [here](https://github.com/maartenba/myget-documentation/pull/1/files#diff-a6e799b6b0d7c47c3631837ea0698fdeR14).

However, I'd add a comment here with anchors to quickly navigate to the client-specific **Private Feed** sections below.

xavierdecoster reviewed (commented):
The image `package-managers/assets/feed-details.png` is using your personal user account, whilst the others are using the `mygetdocs` account. Prefer to make it more consistent `mygetdocs`.

Also, don't forget about `dotnet CLI` and `Paket` :)

maartenba commented on package-managers/nuget.md:
```diff
@@ -1,25 +1,76 @@
+### Visual Studio
+
+#### Public feed
+If you don't know where to find your feed URL check the [Find your feed URL](#find-your-feed-url) section.
```
YOU DON'T KNOW GO AWAY! (is the message given here :-))

Better: `The public feed URL can be found in the feed's details tab. More details can be found in the find your feed URL section.`

maartenba commented on package-managers/nuget.md:
```diff
@@ -1,25 +1,76 @@
+### JetBrains Rider
+
+#### Public feed
+If you don't know where to find your feed URL check the [Find your feed URL](#find-your-feed-url) section.
```
Not sure we need to repeat these sections here. Perhaps have the find you feed URL section from earlier, and here just say "We'll assume we already have the feed's connection details at hand." - which would apply for private and public feeds. Skip the clutter, go to the meat :-)

maartenba reviewed (commented):
Added a few suggestions

warrantor committed:
DOCS-41 Update after comments in PR

maartenba commented on package-managers/nuget.md:
```diff
@@ -6,10 +6,63 @@ TODO (inspiration: http://docs.myget.org/docs/walkthrough/getting-started-with-n
+
+#### Private feed
+
+Private feeds are only accessible to the feed owner, and to whomever the feed owner granted sufficient privileges. When we need to connect to a private feed we don't own, we may need to ask the feed owner to invite us to the feed.
```
`and to whomever the feed owner granted sufficient privileges` -> `and to other users who were granted access by the feed owner` ? (reads a bit more fluent I think)

maartenba commented on package-managers/nuget.md:
```diff
@@ -6,10 +6,63 @@ TODO (inspiration: http://docs.myget.org/docs/walkthrough/getting-started-with-n
+
+Private feeds are only accessible to the feed owner, and to whomever the feed owner granted sufficient privileges. When we need to connect to a private feed we don't own, we may need to ask the feed owner to invite us to the feed.
+
+> ### Hint
```
For tehse hints, check https://raw.githubusercontent.com/maartenba/myget-documentation/d32707bcde4de42162dbe907e061412bcc929228/notesalertscallouts.md on the correct `<div>` tag to use.

maartenba commented on package-managers/nuget.md:
```diff
@@ -6,10 +6,63 @@ TODO (inspiration: http://docs.myget.org/docs/walkthrough/getting-started-with-n
+
+#### Public feed
+
+There are two types of public feeds we can connect to. A public feed we own and one from the gallery. The only real difference here, is where to find the URL of the feed.
```
`There are two types of public feeds we can connect to: public feeds we own, or a feed we want to consume from the gallery. The only real difference is where to find the URL of the feed.`

maartenba commented on package-managers/nuget.md:
```diff
@@ -6,10 +6,63 @@ TODO (inspiration: http://docs.myget.org/docs/walkthrough/getting-started-with-n
+
+There are two types of public feeds we can connect to. A public feed we own and one from the gallery. The only real difference here, is where to find the URL of the feed.
+
+MyGet feeds have multiple endpoints, and each one is specific to a particular package type and protocol. Pick the NuGet endpoint URL that matches the protocol version you want to use (e.g. NuGet v3), and use that to configure the feed endpoint in your favorite NuGet client.
```
Maybe add one minor sentence: `Let's go over how to connect to these.`
In terms of order, preferably first user's own feeds and then second gallery feed?

xavierdecoster replied:
Yep, would swap the order here to user's own feed first.

maartenba commented on package-managers/nuget.md:
```diff
@@ -6,10 +6,63 @@ TODO (inspiration: http://docs.myget.org/docs/walkthrough/getting-started-with-n
+
+#### Public feed
+
+If we don't know where to find our feed URL check the [Find our feed URL](#find-our-feed-url) section.
```
Great use of `we` but since only one sentence, perhaps something like `We will need the URL to our feed, which we can find following the instructions in the find our feed URL section`?

warrantor replied:
`CTRL` + `H` Replace "You" with "We" 😝

maartenba commented on package-managers/nuget.md:
```diff
@@ -18,39 +71,43 @@ could have links to the NuGet documentation
+
+   Unable to find version xxxx of package yyyy
+
+The reason this happens is because the NuGet command line, the NuGet Visual Studio Extension and the NuGet PoourShell Console all have a configuration option specifying which package source to install from. When this setting is changed to one specific feed, other feeds will be ignored and the error above will be shown during package restore.
```
`PoourShell` ? :-)

maartenba commented on package-managers/nuget.md:
```diff
@@ -18,39 +71,43 @@ could have links to the NuGet documentation
+
+The reason this happens is because the NuGet command line, the NuGet Visual Studio Extension and the NuGet PoourShell Console all have a configuration option specifying which package source to install from. When this setting is changed to one specific feed, other feeds will be ignored and the error above will be shown during package restore.
+
+The solution is very simple: we can set the active package source to aggregate in Visual Studio, or simply configure NuGet to always use the aggregate package source for the current project. NuGet has an inheritance system for NuGet.config files, where the NuGet.config file closest to the solution file gets the last say. If we add the following NuGet.config file next to the solution file for our  ...
```
add quotes, e.g. `"aggregate"`

maartenba commented on package-managers/nuget.md:
````diff
@@ -18,39 +71,43 @@ could have links to the NuGet documentation
+    </configuration>
+```
+
+We can take this one step further and instead of configuring our MyGet feed globally for our system (and requiring other devs on our team to do the same), why not distribute a NuGet.config along with the sources?
````
Add: We can place a `NuGet.config` file right next to our `.sln` file for Visual Studio to pick it up.

xavierdecoster replied:
Actually, it's *NuGet* to pick it up ;)

maartenba reviewed (commented):
Thanks man! Added a few nits that may need adressing.

xavierdecoster reviewed (commented):
Good stuff!

Don't forget about the `package-managers/assets/feed-details.png` image, which still uses your personal account instead of the `mygetdocs` account.

Commit:
DOCS-41 Implemented extra comments on PR

Switched order to start with private feeds.
Changed 'Hints' to HTML 'Tip'.
Updated content.
Replace poourshell with powershell.
Replaced galery feed image

maartenba commented on package-managers/nuget.md:
```diff
@@ -6,10 +6,65 @@ TODO (inspiration: http://docs.myget.org/docs/walkthrough/getting-started-with-n
+
+#### Private feed
+
+Private feeds are only accessible to the feed owner, and to other users who were granted access by the feed owner. When we need to connect to a private feed we don't own, we may need to ask the feed owner to invite us to the feed.
```
`We` is good, but a bit over the top here :-) Consider: "To connect to a feed we don't own, we may have to request permissions from the feed owner." ?

maartenba commented on package-managers/nuget.md:
```diff
@@ -6,10 +6,65 @@ TODO (inspiration: http://docs.myget.org/docs/walkthrough/getting-started-with-n
+Private feeds are only accessible to the feed owner, and to other users who were granted access by the feed owner. When we need to connect to a private feed we don't own, we may need to ask the feed owner to invite us to the feed.
+
+<p class="alert alert-success">
+    <strong>Tip:</strong> We can find out how to invite users <a href="../inviting-users-to-a-feed.md">here</a>
```
`We can` -> just `Fnd out how to ....` - no need for we can here

maartenba commented on package-managers/nuget.md:
```diff
@@ -6,10 +6,65 @@ TODO (inspiration: http://docs.myget.org/docs/walkthrough/getting-started-with-n
+![Feed details](assets/feed-details.png)
+
+<p class="alert alert-success">
+    <strong>Tip:</strong> We support <a href="#visualstudio">Visual Studio</a>, <a href="#rider">JetBrains Rider</a>, the <a href="#nugetcli">NuGet CLI</a>, <a href="#dotnetcli">dotnet CLI</a> and <a href="#paket">Paket</a>
```
`, Paket and any other NuGet clients that use the standard NuGet v2 or v3 protocols`.

maartenba commented on package-managers/nuget.md:
```diff
@@ -6,10 +6,65 @@ TODO (inspiration: http://docs.myget.org/docs/walkthrough/getting-started-with-n
+
+After browsing the list of feeds on the [MyGet Gallery](https://myget.org/gallery) (or on the MyGet Enterprise Gallery), we can get its connection details using the **Connect to feed** button. This will open a dialog where we can find the various endpoints for this feed.
+
+![Gallery feed details](assets/gallery-feed.png)
```
Maybe add a little bit of text between the two images? Just to make it less dense, visually.

maartenba commented on package-managers/nuget.md:
```diff
@@ -6,10 +6,65 @@ TODO (inspiration: http://docs.myget.org/docs/walkthrough/getting-started-with-n
+
+#### Public feed
+
+If we don't know where to find our feed URL check the [Find our feed URL](#find-our-feed-url) section.
```
"If we don't know" <-- are you accusing our users of not knowing things! (maybe a looser `To find the feed URL, check the find our feed URL section`)

maartenba commented on package-managers/nuget.md:
```diff
@@ -6,10 +6,65 @@ TODO (inspiration: http://docs.myget.org/docs/walkthrough/getting-started-with-n
+
+#### Public feed
+
+If we don't know where to find our feed URL check the [Find our feed URL](#find-our-feed-url) section.
```
Same here

maartenba commented on package-managers/nuget.md:
```diff
@@ -6,10 +6,65 @@ TODO (inspiration: http://docs.myget.org/docs/walkthrough/getting-started-with-n
+
+#### Public feed
+
+If we don't know where to find our feed URL check the [Find our feed URL](#find-our-feed-url) section.
```
And here

maartenba commented on package-managers/nuget.md:
```diff
@@ -6,10 +6,65 @@ TODO (inspiration: http://docs.myget.org/docs/walkthrough/getting-started-with-n
+
+#### Private feed
+
+### NuGet CLI <a id="nugetcli"></a>
```
No need for the `<a id`, I think anchors are auto-generated based on the topic title (this one will be `#nuget-cli` I believe)

warrantor replied:
I did this here because I am calling it from HTML, it didn't work in my preview screen, will it automagically work on gitbook?

maartenba commented on package-managers/nuget.md:
````diff
@@ -20,37 +75,40 @@ could have links to the NuGet documentation
+    </configuration>
+```
+
+We can take this one step further and instead of configuring our MyGet feed globally for our system (and requiring other devs on our team to do the same), why not distribute a NuGet.config along with the sources? We can place a `NuGet.config` file right next to our `.sln` file for NuGet to pick it up.
````
Add "`"  around buth NuGet.config references, now it's just the second one.

maartenba reviewed (commented):
Reviewed

Commit:
DOCS-41 Updated after comments
Will add new content after this PR is approved.

maartenba reviewed (commented):
LGTM :) @xavierdecoster  ?

xavierdecoster commented:
Text LGTM, screenshots should use consistent, non-personal accounts imho. Can we use the `mygetdocs` account everywhere?
